core/unit_tests/_testing.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Shared testing utilities."""
+
+
+class _Monkey(object):
+    # context-manager for replacing module names in the scope of a test.
+
+    def __init__(self, module, **kw):
+        self.module = module
+        if len(kw) == 0:  # pragma: NO COVER
+            raise ValueError('_Monkey was used with nothing to monkey-patch')
+        self.to_restore = {key: getattr(module, key) for key in kw}
+        for key, value in kw.items():
+            setattr(module, key, value)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for key, value in self.to_restore.items():
+            setattr(self.module, key, value)
+
+
+class _NamedTemporaryFile(object):
+
+    def __init__(self, suffix=''):
+        import os
+        import tempfile
+        filehandle, self.name = tempfile.mkstemp(suffix=suffix)
+        os.close(filehandle)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        import os
+        os.remove(self.name)
```

Comment:
This is just the new GitHub UI playing tricks on you. This file is gone. From a fresh checkout:

```
$ git checkout make-core-subpackage
Branch make-core-subpackage set up to track remote branch make-core-subpackage from origin.
Switched to a new branch 'make-core-subpackage'
$ git ls-files '*_testing.py'
core/google/cloud/_testing.py
unit_tests/bigtable/_testing.py
```

I did #2394 so I could just safely move the `_testing` module in one of my CLI-only commits.

The commit history should be easy to grok. There are 2 massive rename-only commits (with the rename script in the commit message) and then the rest are very small and focused.